semantics/src/main/kotlin/com/strumenta/kolasu/semantics/scope/provider/declarative/DeclarativeScopeProvider.kt

```diff
@@ -46,16 +48,19 @@ inline fun <
  * ```
  **/
 open class DeclarativeScopeProvider(
+    open var nodeIdProvider: NodeIdProvider = StructuralNodeIdProvider(),
```

Comment:
I think this is now unused and could be removed

Reply:
You are right: removing it (and remove it from the DeclarativeScopeProviderRule, as well, because otherwise code did not compile) everything still seems to work